unyt/array.py

```diff
@@ -1581,6 +1582,18 @@ def ua(self):
         """
         return np.ones_like(self)
 
+    @property
+    def name(self):
+        """Return the name (string) of the array"""
+        name = getattr(self, "_name", "")
+        if name is None:
+            name = ""
+        return name
+
+    @name.setter
+    def name(self, name):
+        self._name = name
```

Comment:
Why does this need to be a property? I think it's simpler just to make this an attribute and delete the custom getter/setter unless you want to do some additional checking in the setter. I guess you could check if the provided name is really a string and if it's not then raise an error.

Reply:
In my original implementation, ``name`` wasn't in the constructor hence the property.